tests/test_correct_fsr_volume/test_correct_fsr_volume.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+import os
+import sys
+sys.path.insert(0, os.pardir)
+sys.path.insert(0, os.path.join(os.pardir, 'openmoc'))
+from testing_harness import TestHarness
+from input_set import PinCellInput
+
+
+class CorrectFSRVolumeTestHarness(TestHarness):
+    """Tests Geometry::correctFSRVolume() method."""
+
+    def __init__(self):
+        super(CorrectFSRVolumeTestHarness, self).__init__()
+        self.input_set = PinCellInput()
+        self._result = ''
+
+    def _generate_tracks(self):
+        """Generate Tracks and segments and correct the FSR volume for
+        the fuel."""
+
+        # Always use 1 thread for FSR reproducibility
+        self.track_generator.setNumThreads(1)
+        self.track_generator.generateTracks()
+
+        old_volume = self.track_generator.getFSRVolume(1)
+        self.track_generator.correctFSRVolume(1, 2.7)
+        one_thread_volume = self.track_generator.getFSRVolume(1)
+
+        # Save results for one thread
+        self._result += str(round(old_volume,6)) + '\n'
+        self._result += str(round(one_thread_volume, 1)) + '\n'
+
+        # get results with multiple threads
+        self.track_generator.setNumThreads(2)
+        self.track_generator.correctFSRVolume(1, 2.5)
```

Comment:
I used different volumes just to ensure there isn't a cases where multithreading computes the old volume. I like your solution better, so I'm make the change.